### PR TITLE
Add heartbeat support

### PR DESCRIPTION
**Pull request by makinoshi**
Add heartbeat support.

If `heartbeat` and `heartbeat-timeout` option is set, the consumer extend message visibility per `hearbeat` seconds to `heartbeat-timeout` seconds.

### CODE REVIEW

File: test/gluttony/core_test.clj [hunk 1]
```diff
@@ -117,4 +124,35 @@
           (a/<!! (th/wait-chan (* 1000 45) (fn [] (>= (count @collected) 20))))
           (is (= (set (range 1 21))
                  (set @collected)))
+          (stop-consumer consumer)))
+
+      (testing "Gather every data in order"
```
Comment (liquidz) [inline test/gluttony/core_test.clj]: The label for this test case seems to be mismatched to what you are testing.
It should be recognizable as a `heartbeat` test.

**Review by liquidz — CHANGES_REQUESTED** [review]
Overall the PR looks good to me ☺️ 
Could you change only the label for `heartbeat` testing?

File: test/gluttony/core_test.clj (hunk 1 above)
Comment (makinoshi) [reply test/gluttony/core_test.clj]: Thanks!

**Comment by makinoshi** [comment]
@liquidz Thank you for your review. I've fixed your point.

**Review by liquidz — APPROVED** [review]
LGTM

**Comment by makinoshi** [comment]
🙏

**Pull request merged**